test/table.ts

```diff
@@ -2080,215 +2128,198 @@ describe('BigQuery/Table', () => {
 
     it('should execute callback with error & API response', done => {
       const error = new Error('Error.');
-      const apiResponse = {};
-
-      table.request = (reqOpts: DecorateRequestOptions, callback: Function) => {
-        callback(error, apiResponse);
-      };
+      requestStub.rejects(error);
 
       table.insert(data, (err: Error, apiResponse_: {}) => {
         assert.strictEqual(err, error);
-        assert.strictEqual(apiResponse_, apiResponse);
```

Comment:
Do we lose getting back the "apiResponse" argument when there's an error? Previously, we returned the "err" and "resp" from both `this.create()` and `this.request()` when they returned an error.

Reply:
Yes, promise rejections only allow a single error object. The api response could be added as a property of the error object, but that is not a pattern used in these node repos. This was due to the additional abstractions provided early-on in this PR for asserting the table, and handling the retries.